app/controllers/contacts_controller.rb

```diff
@@ -68,6 +70,16 @@ def update
     invalid_update
   end
 
+  def name_for_lead_service(role_id)
```

Comment:
Does the role_id need to be passed in?
Could just be:
```
  Roles.find(lead_service_id).name if lead_service_id
```

Reply:
or `lead_service_id.present? ? Roles.find(lead_service_id).name : 'No lead service'` etc.

Reply:
I am not sure how rails handles that but I was hoping to have a cached list of roles instead of querying it for every row, I thought it may be a bit faster

Reply:
so basically instead of querying

findById 10 times, it only query findAll roles 1 time and filter that list, I am not sure if the performance is noticeable